### PR TITLE
[E2e] Get logs of containers that have restarted

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Print the last log lines of containers that have restarted during the execution of containers e2e tests.

### Motivation

It sometimes happens that a container restarts during the execution of containers e2e tests.
If it’s a fake app workload, it can be a clue for a failing test.
If it’s an agent container, it will make the `Test00UpAndRunning` test fail.

So far, the best way to investigate such container restarts was too look at their logs on the `dddev` org as the agent dual-ships all the collected data there.
Unfortunately, there are some cases where this doesn’t work and we can’t find the logs of the restarted container:
* When the container has restarted before the datadog agent was up and running.
* When the datadog agent itself is in `CrashLoopBackOff`.

For those situations, it would be valuable to directly collect the container’s logs from k8s.

### CODE REVIEW

File: test/new-e2e/tests/containers/dump_cluster_state.go [hunk 1]
```diff
@@ -286,4 +289,55 @@ func dumpK8sClusterState(ctx context.Context, kubeconfig *clientcmdapi.Config, o
...
+		return
+	}
+
+	continueToken := "start"
+	for continueToken != "" {
+		if continueToken == "start" {
+			continueToken = ""
+		}
+		pods, err := k8sClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+			Continue: continueToken,
```
Comment (vboulineau) [inline test/new-e2e/tests/containers/dump_cluster_state.go:311]: Is this required? I thought that by default it returns everything?

File: test/new-e2e/tests/containers/dump_cluster_state.go [hunk 2]
```diff
@@ -286,4 +289,55 @@ func dumpK8sClusterState(ctx context.Context, kubeconfig *clientcmdapi.Config, o
...
+		continueToken = pods.Continue
+
+		for _, pod := range pods.Items {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.RestartCount > 0 {
+					fmt.Fprintf(out, "\nLOGS FOR POD %s/%s CONTAINER %s:\n", pod.Namespace, pod.Name, containerStatus.Name)
+					logs, err := k8sClient.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+						Container: containerStatus.Name,
+						Previous:  true,
+						TailLines: pointer.Ptr(int64(20)),
```
Comment (vboulineau) [inline test/new-e2e/tests/containers/dump_cluster_state.go:326]: Why not all? We don't expect loads of logs

Comment (L3n41c) [reply test/new-e2e/tests/containers/dump_cluster_state.go:326]: I feared spamming the GitLab console if a verbose container (an agent with DEBUG log level or an nginx logging every single query) was OOM killed after 20 minutes.
But I can put a higher reasonable limit.

File: test/new-e2e/tests/containers/dump_cluster_state.go (hunk 1 above)
Comment (L3n41c) [reply test/new-e2e/tests/containers/dump_cluster_state.go]: If the default limit is “unlimited”, let’s get rid of paging support.

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**